fenix/app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -1062,7 +1062,7 @@ abstract class BaseBrowserFragment :
             if (context.settings().shouldUseBottomToolbar) {
                 swipeRefreshParams.bottomMargin = toolbarHeight
             } else {
-                swipeRefreshParams.topMargin = toolbarHeight
+                swipeRefreshParams.topMargin = 0
```

Comment:
This seems to cause issues in normal browsing:

![image](https://user-images.githubusercontent.com/11428869/220120364-a6b1e321-8ae2-4dc5-8c29-a8815581d328.png)

Seems like we need to differentiate between the different scenarios and pass the right `toolbarHeight` value to `initializeEngineView` which would directly set it.